cmd/log-collector/cmd/helper.go

```diff
@@ -0,0 +1,164 @@
+		logCollector.Loglevel = logLevel
+	}
+	if cmd.Flags().Changed(clusteredFlag) {
+		logCollector.Clustered = clustered
+	}
+	if cmd.Flags().Changed(namespacesFlag) {
+		logCollector.Namespaces = namespaces
+	}
+	if cmd.Flags().Changed(keepSourceFlag) {
+		logCollector.CleanOutput = keepSource
+	}
+	if cmd.Flags().Changed(gvkFlag) {
+		gvks, err := parseGVK(gvkSlice)
+		if err != nil {
+			return err
+		}
+		logCollector.GroupVersionKinds = deDuplicateGVKs(gvks)
+	}
+	if cmd.Flags().Changed(labelsFlag) {
+		labels, err := parseLabelSelector(labelSlice)
+		if err != nil {
+			return err
+		}
+		logCollector.LabelSelectors = deDuplicateLabelSelector(labels)
+	}
+
+	return nil
+}
+
+func parseGVK(gvkSlice []string) ([]logcollector.GroupVersionKind, error) {
+	var gvks []logcollector.GroupVersionKind
+	groupList, err := logCollector.DisClient.ServerGroups()
+
+	if err != nil {
+		if !discovery.IsGroupDiscoveryFailedError(err) {
+			log.Error(err, "Error while getting the resource list from discovery client")
+			return gvks, err
+		}
+		log.Warnf("The Kubernetes server has an orphaned API service. Server reports: %s", err.Error())
+		log.Warn("To fix this, kubectl delete apiservice <service-name>")
+	}
+
+	for idx := range gvkSlice {
+		splitGVK := strings.Split(gvkSlice[idx], "/")
+		if len(splitGVK) == 3 && splitGVK[2] != "" {
+			if splitGVK[1] == "" {
+				for idx := range groupList.Groups {
+					if strings.EqualFold(groupList.Groups[idx].Name, splitGVK[0]) {
+						splitGVK[1] = groupList.Groups[idx].PreferredVersion.Version
+					}
+				}
+			}
+			gvk := logcollector.GroupVersionKind{
+				Group:   splitGVK[0],
+				Version: splitGVK[1],
+				Kind:    splitGVK[2],
+			}
+			gvks = append(gvks, gvk)
+		} else {
+			if splitGVK[2] == "" {
+				return gvks, errors.New("kind cannot be empty in gvks flag ")
+			}
+			return gvks, errors.New("error parsing gvks ")
+		}
+	}
+	return gvks, nil
+}
+
+func parseLabelSelector(labelSlice []string) ([]apiv1.LabelSelector, error) {
+	var lbSelectors []apiv1.LabelSelector
+
+	for idx0 := range labelSlice {
+		andLabels := strings.Split(labelSlice[idx0], ",")
+		matchLabels := make(map[string]string)
+		for idx := range andLabels {
+			mapKeysValues := strings.Split(andLabels[idx], "=")
+			if len(mapKeysValues) == 2 {
+				matchLabels[mapKeysValues[0]] = mapKeysValues[1]
+			} else {
+				return lbSelectors, errors.New(" Error Parsing Labels ")
+			}
+		}
+		labelSelector := apiv1.LabelSelector{
+			MatchLabels: matchLabels,
+		}
+		lbSelectors = append(lbSelectors, labelSelector)
+	}
+	return lbSelectors, nil
+}
+
+func deDuplicateLabelSelector(lbSelectors []apiv1.LabelSelector) []apiv1.LabelSelector {
+	var uniqueSelectors []apiv1.LabelSelector
+
+	for idx := range lbSelectors {
+		skip := false
+		for idx1 := range uniqueSelectors {
+			if reflect.DeepEqual(lbSelectors[idx], uniqueSelectors[idx1]) {
+				skip = true
+				break
+			}
+		}
```

Comment:
create `set` and use it to check if current gvk/label is already inserted and  `lbSelectors[idx].String()` can give string value to insert into set